tests/core/pyspec/eth2spec/test/gloas/block_processing/test_process_execution_payload_bid.py

```diff
@@ -348,6 +354,48 @@ def test_process_execution_payload_bid_slashed_builder(spec, state):
 
     yield from run_execution_payload_bid_processing(spec, state, block, valid=False)
 
+@with_gloas_and_later
+@spec_state_test
+def test_process_block_then_slash_non_timely_builder(spec, state):
+    """
+    Empty-case payment path:
+    - A builder posts a successful bid in a block, but the payload is not revealed in time.
+    - Builder is slashed
+    - Proposer unconditional payment is enforced by moving the bid amount into
+    """
+
+    # Get past genesis
+    next_epoch(spec, state)
+    next_epoch(spec, state)
+
+    block, builder_index = prepare_block_with_non_proposer_builder(spec, state)
+
+    value = spec.Gwei(1_000_000)
+    state.balances[builder_index] = value + spec.MIN_ACTIVATION_BALANCE
+
+    # Create a bid with value amount
+    signed_bid = prepare_signed_execution_payload_bid(
+        spec,
+        state,
+        builder_index=builder_index,
+        value=value,
+        slot=block.slot,
+        parent_block_root=block.parent_root,
+    )
+
+    block.body.signed_execution_payload_bid = signed_bid
+    yield "pre", state
+    yield "block", block
+    spec.process_block(state, block)
+    # Slash validator
+    spec.slash_validator(state, builder_index)
```

Comment:
This state change would have to be after the post, but I think it is better to use another format that sorrounds the slashing